dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -130,6 +134,32 @@ public void run(@Nonnull ConnectorArguments arguments) throws Exception {
   private <T> T runTask(
       @Nonnull TaskRunContext context, @Nonnull TaskSetState.Impl state, Task<T> task)
       throws MetadataDumperUsageException {
+    T t = executeTask(context, state, task);
+
+    numberOfCompletedTasks += 1;
```

Comment:
The `+=` operation is not atomic. Since the tasks can be executed in parallel, then I would suggest to turn `numberOfCompletedTasks` into `AtomicInteger`.